askama_shared/src/derive.rs

```diff
@@ -0,0 +1,107 @@
+extern crate proc_macro;
```

Comment:
So I tried this in the past and this doesn't work. As far as I remember, you should not import `proc_macro` anywhere except in a `proc-macro = true` crate. However, what we can do is convert to proc_macro2 types in askama_derive and then pass that into askama_shared.

I don't love the `derive` name of this module, given that I think we might want an attribute macro at some point in the not too distant future, too. Can we use `macro` as the name, or is that a keyword?

Reply:
I replaced proc_macro by proc_macro2.

Actually, I guess I could move everything into generator.rs. It's only 100 lines as is.

Reply:
> Actually, I guess I could move everything into generator.rs. It's only 100 lines as is.

That probably makes sense for now, although it just means we should start splitting that up since it's already pretty big. Good start, though!